Update installation guide to match how-to template (#184)
napari/napari.github.io: tutorials/fundamentals/installation.md

# napari installation tutorial
# How to install napari on your machine

Welcome to the **napari** installation tutorial!
Welcome to the **napari** installation guide!

This tutorial will teach you how to do a clean install of **napari**. It is
aimed at people that just want to use napari. For people also interested in
contributing to napari please check our [contributing

> guidelines](https://github.com/napari/napari/blob/master/docs/developers/CONTRIBUTING.md)

Reviewer: I think we may want to keep this redirection for people looking for the developer installation guide instead

Reviewer: Ah, that link doesn't work anyway. Here it is: https://napari.org/developers/contributing.html

Author: Yep, I've added this at the very top in a note:

```{note} 
If you want to contribute back to the napari codebase, you should follow [this guide instead](https://napari.org/developers/contributing.html).
```

### Installing the bundled app
This is the easiest way to install napari if you only wish to use it as a standalone GUI app.
This installation method does not have any prerequisites. 
[Click here](#installing-as-a-bundled-app) to see instructions

> for installing the bundled app.

Reviewer: Possibly we should talk about the restrictions involved with choosing the bundled app over the other methods

Author: @GenevieveBuckley I added an important note above:

```{important}
Note that the bundled app is still
in active development, and may not be very stable. We strongly recommend installing as a Python
package instead.
```

Is this sufficient mention of the restrictions?

Reviewer: Yes, I think so. Might be good to link to the python installation section/instructions in that sentence, too.

> ### Installing as a python package or in editable mode

Reviewer: Can we flag this as the method as "recommended"?

The bundle is still a bit experimental, and it's not so easy for people to add new packages to the bundle environment, etc. I think generally we prefer if people use napari installed as a python package.

This installation method allows you to use napari from Python to programmatically 
interact with the app.

It requires:
- [Python 3.7+](https://www.python.org/downloads/)
- the ability to install python packages via [pip](https://pypi.org/project/pip/) OR [conda-forge](https://conda-forge.org/docs/user/introduction.html)

You may also want:
- an environment manager like [conda](https://docs.conda.io/en/latest/miniconda.html) or
[venv](https://docs.python.org/3/library/venv.html) **(Highly recommended)**
- [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) (For editable mode)

***

> ### Choose which distribution to install.

Reviewer 2: ```suggestion
## Choose which distribution to install.
```
Going up one level ensures people understand the next sections are options and they should only go with one of them. I'd go as far as using numbering to make that even clearer:

1. From pip
2. From conda-forge
3. From the main branch on GitHub

Author: Yeah I think it would be nice if this was a dropdown selection similar to [this section](https://numpy.org/numpy-tutorials/content/pairing.html#pair-your-notebook-files-ipynb-and-md) in the numpy notebook/jupytext guide:

1. It would limit the amount of code those who want to install the bundled app see at first glance if scrolling through the document
2. It would limit distractions for those who just want to install via pip or via conda - you probably don't want to do all of them and if you do, it doesn't feel like that much effort to click all three
3. It clearly shows that these are three separate options and in general you would pick just one.

Thoughts? I'm not sure how easy this would be to implement given the current theme because I don't remember seeing any drop down/accordion behaviour. Pinging @codemonkey800 - do we have something like this? If not, could we bring over something like the [FAQ dropdown](https://www.napari-hub.org/faq) from the hub?

Reviewer 2: That is a [jupyter-book feature](https://jupyterbook.org/interactive/hiding.html) so might be already doable - although I don't know about the styling using the napari theme.

Reviewer 3: The napari tutorials use this feature extensively, so we'll also have support for it in the napari theme 😄 

You can see a demo of it in #179.

---
jupyter:
  jupytext:
    formats: ipynb,md
    text_representation:
      extension: .md
      format_name: markdown
      format_version: '1.3'
      jupytext_version: 1.13.0
  kernelspec:
    display_name: Python 3 (ipykernel)
    language: python
    name: python3
---

# How to install napari on your machine

Welcome to the **napari** installation guide!

This guide will teach you how to do a clean install of **napari** and launch the viewer.

```{note} 

> If you want to contribute back to the napari codebase, you should follow [this guide instead](https://napari.org/developers/contributing.html).

Reviewer: ```suggestion
If you want to contribute code back into napari, you should follow the [development installation instructions in the contributing guide](https://napari.org/developers/contributing.html) instead.
```

I'd kind of like the words "development installation" included in this sentence, otherwise it's a bit hard to see how this page content relates to this note.

```

## Prerequisites

Prerequisites differ depending on how you want to install napari.

> ### Installing as a Python package 

Reviewer: ```suggestion
### Prerequisites for installing napari as a Python package 
```

This installation method allows you to use napari from Python to programmatically 
interact with the app. It is the best way to install napari and make full use of
all its features.

It requires:

> - [Python 3.7+](https://www.python.org/downloads/)

Reviewer: ```suggestion
- [Python 3.7 or higher](https://www.python.org/downloads/)
```

Just for clarity :smile:

Reviewer: Although, I'm not sure you should link to https://www.python.org/downloads/ if you are later going to try to encourage the use of conda. Maybe the link can be removed?

- the ability to install python packages via [pip](https://pypi.org/project/pip/) OR [conda-forge](https://conda-forge.org/docs/user/introduction.html)

You may also want:
- an environment manager like [conda](https://docs.conda.io/en/latest/miniconda.html) or
[venv](https://docs.python.org/3/library/venv.html) **(Highly recommended)**

> ### Installing the bundled app

Reviewer: ```suggestion
### Prerequisites for installing napari as a bundled app
```

This is the easiest way to install napari if you only wish to use it as a standalone GUI app.
This installation method does not have any prerequisites. 

```{important}
Note that the bundled app is still
in active development, and may not be very stable. We strongly recommend 
[installing as a Python package instead](#choose-which-distribution-to-install).
```

[Click here](#installing-as-a-bundled-app) to see instructions
for installing the bundled app.

> ## Install as Python package

Reviewer: ```suggestion
## Install as Python package (recommended)
```

Python package distributions of napari can be installed via `pip`, `conda-forge`, or from source.

````{important}

> While not strictly required, it is *highly* recommended to install

Reviewer: ```suggestion
While not strictly required, it is highly recommended to install
```

Personal preference, I'd prefer we didn't have such a mix of emphasis markers so close together. (There is bold text above for **(Highly recommended)**, the "Important" box is also for emphasis, I don't think we need a single word in italics to further emphasize things on top of all that)

napari into a clean virtual environment using an environment manager like

> [conda](https://docs.conda.io/en/latest/miniconda.html) or

Reviewer: This is my favourite docs page on how to use conda environments, maybe it's worthwhile linking to it here? Up to you.
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html

Author: Lovely, thanks Gen!

[venv](https://docs.python.org/3/library/venv.html).

This should be set up *before* you install napari. For example, with `conda`:

```sh
conda create -y -n napari-env -c conda-forge python=3.9
conda activate napari-env
```
````

Choose one of the options below to install napari as a Python package.

````{admonition} **1. From pip**
:class: dropdown

napari can be installed on most macOS, Linux, and Windows systems with Python
3.7, 3.8, and 3.9 using pip:

```sh

> pip install "napari[all]"

Reviewer: ```suggestion
python -m pip install "napari[all]"
```

Suggestion: we should consider making it a general rule that anytime we have instructions with pip, we show `python -m pip ...`. It reduces the chance that someone unfamiliar with virtual environments will end up pip installing packages into their base environment instead by mistake, etc.

```
You can then upgrade napari to a new version using

```sh

> pip install "napari[all]" --upgrade

Reviewer: ```suggestion
python -m pip install "napari[all]" --upgrade
```

---
jupyter:
  jupytext:
    formats: ipynb,md
    text_representation:
      extension: .md
      format_name: markdown
      format_version: '1.3'
      jupytext_version: 1.13.0
  kernelspec:
    display_name: Python 3 (ipykernel)
    language: python
    name: python3

> ---

Reviewer 2: Do we want to make this document executable?

Author: I've made this document executable for now purely so I can use the Jupyter book dropdown feature, and also because I think in general we want all tutorials to be MyST markdown eventually. 
Once we merge the dropdown accordion feature (issue #279) we can change this dropdown to match that styling. Do you think there are any dangers to keeping it executable until then @melissawm?

```

## Prerequisites

Prerequisites differ depending on what type of distribution you want to install.

### Installing the bundled app
This is the easiest way to install napari if you only wish to use it as a standalone GUI app.
This installation method does not have any prerequisites. 

```{important}
Note that the bundled app is still
in active development, and may not be very stable. We strongly recommend installing as a Python
package instead.
```

[Click here](#installing-as-a-bundled-app) to see instructions
for installing the bundled app.

### Installing as a python package 
This installation method allows you to use napari from Python to programmatically 
interact with the app. It is the best way to install napari and make full use of
all its features.

It requires:
- [Python 3.7+](https://www.python.org/downloads/)
- the ability to install python packages via [pip](https://pypi.org/project/pip/) OR [conda-forge](https://conda-forge.org/docs/user/introduction.html)

You may also want:
- an environment manager like [conda](https://docs.conda.io/en/latest/miniconda.html) or
[venv](https://docs.python.org/3/library/venv.html) **(Highly recommended)**

> ## Choose which distribution to install

Reviewer 2: I feel like we have redundant information here, since the different installation methods have already been mentioned above. Maybe the "Choose which distribution to install" section could come before the particular installation methods.

Author: Yeahh this one is a bit weird because "Choose which distribution to install" does only refer to installing as package - it's just do you want it from `pip`, `conda` or `github main branch`. I'll change the wording here to reflect this a bit better.

conda install -c conda-forge napari
```

> ### Install from the master branch on Github

Reviewer 2: ```suggestion
### Install from the main branch on Github
```

> To install the "next-release" version from github via pip, call

Reviewer: ```suggestion
To install the "bleeding edge" version from github via pip, call
```

Reviewer: I don't think "next-release" is very clear (but "bleeding edge" is also not very clear, so if you have any better ideas we should do that)

Reviewer 2: Development version? Unstable version?

```sh
pip install "git+https://github.com/napari/napari.git#egg=napari[all]"
```

> ### Clone the repository locally and install in editable mode

Reviewer: I don't think this content should live here - editable pip installations are for the developer docs (and these instructions are also slightly different to what's written there https://napari.org/developers/contributing.html)

conda install -c conda-forge napari
```

You can then upgrade to a new version of napari using 

```sh
conda update napari
```
````

````{admonition} **3. From the main branch on Github**
:class: dropdown

To install the latest version with yet to be released features from github via pip, call

```sh

> pip install "git+https://github.com/napari/napari.git#egg=napari[all]"

Reviewer: ```suggestion
python -m pip install "git+https://github.com/napari/napari.git#egg=napari[all]"
```

The very first time you launch napari the startup time will be fairly slow, but
after that you will find that napari launches more quickly.

### Installing the Linux bundle

(Guide coming soon... In the meantime, if you try it and encounter issues, see
below for how to contact us.)

> ***

Reviewer: What's this line about? Was there supposed to be bold text here?

Author: this was to include a horizontal line separator between the content headings and the template headings, but I think stuff like that is supposed to be dictated by the theme instead.

The very first time you launch napari the startup time will be fairly slow, but
after that you will find that napari launches more quickly.

### installing the Linux bundle
### Installing the Linux bundle

(Guide coming soon... In the meantime, if you try it and encounter issues, see
below for how to contact us.)
<!-- #endregion -->

## bug reports

If you are running into issues, please open a new issue on our [issue

> tracker](https://github.com/napari/napari/issues) and include the output of the

Reviewer: Did we lose this content on getting help if needed, or has it just been moved elsewhere in the document? I think the link for where to submit an issue is useful information.

Author: this has been moved to the bottom under "Next steps".
